File: mobile-conductor/lib/features/balances/data/mappers/pago_balance_mapper.dart
import '../../domain/entities/pago_balance.dart';

class PagoBalanceMapper {
  static PagoBalance fromJson(Map<String, dynamic> json) {
    return PagoBalance(
      id: json['id'] as String,
      viajeId: json['viajeId'] as String,
      montoBruto: (json['montoBruto'] as num).toDouble(),
      feeProcesamiento: (json['feeProcesamiento'] as num).toDouble(),
      montoNeto: (json['montoNeto'] as num).toDouble(),
      metodo: json['metodo'] as String? ?? 'Efectivo',
      fecha: DateTime.tryParse(json['fecha']?.toString() ?? '') ?? DateTime.now(),
    );
  }
}

File: backend/src/features/pagos-balances/aplicacion/mappers/pago-balance.mapper.ts
import { PagoBalance } from '../../dominio/entidades/pago-balance.entity.js';
import { PagoBalanceResponseDto } from '../dto/pago-balance-response.dto.js';

export class PagoBalanceMapper {
  static toResponse(pago: PagoBalance): PagoBalanceResponseDto {
    return {
      id: pago.id,
      viajeId: pago.viajeId,
      conductorId: pago.conductorId,
      montoBruto: pago.montoBruto,
      feeProcesamiento: pago.feeProcesamiento,
      montoNeto: pago.montoNeto,
      metodo: pago.metodo,
      fecha: pago.fecha,
    };
  }
}
